python/ray/data/grouped_data.py

```diff
@@ -539,6 +539,72 @@ def std(
         """
         return self._aggregate_on(Std, on, ignore_nulls=ignore_nulls, ddof=ddof)
 
+    @PublicAPI(api_group=CDS_API_GROUP)
+    def random_sample(
+        self,
+        frac: float,
+        *,
+        random_state: Optional[int] = None,
+    ) -> "Dataset":
+        """Returns a new :class:`Dataset` containing a random fraction of the rows for each group.
+
+        .. note::
+
+            This method returns roughly ``frac * total_rows`` rows per group. An exact number
+            of rows isn't guaranteed.
+
```

Comment:
```suggestion
.. note::
    This method does not shuffle elements within each group nor between groups.
```
Adding one more note to the doc..